aptos-move/aptos-aggregator/src/aggregator_extension.rs

```diff
@@ -13,22 +16,50 @@ use move_binary_format::errors::{PartialVMError, PartialVMResult};
 use move_core_types::account_address::AccountAddress;
 use std::collections::{BTreeMap, BTreeSet};
 
+/// Describes how the `speculative_start_value` in
+/// `AggregatorState` is obtained.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum SpeculativeStartValue {
+    // The speculative_start_value is assigned to 0.
+    Unset,
+    // The speculative_start_value is obtained by reading
+    // the last committed value of the aggregator from MVHashmap.
+    LastCommittedValue(u128),
+    // The speculative_start_value is obtained by aggregating all
+    // the previous deltas of the aggregator from MVHashmap that
+    // were present at the fetch time.
+    AggregatedValue(u128),
+}
+
+impl SpeculativeStartValue {
+    pub fn get_value(&self) -> u128 {
+        match self {
+            SpeculativeStartValue::Unset => 0,
```

Comment:
This can be dangerous, I believe this method should be returning Result<u128> and have the caller explicitly expect, to make sure it's not unset when it's supposed to be set.